lib/flowViewport/sceneIndex/fvpLightsManagementSceneIndex.cpp

```diff
@@ -0,0 +1,163 @@
+    HdContainerDataSourceEditor editor(prim.dataSource);
+    editor.Set(
+        HdLightSchema::GetDefaultLocator().Append(HdLightTokens->ambient),
+        HdRetainedTypedSampledDataSource<float>::New(0.0f)
+    );
+    editor.Set(
+        HdLightSchema::GetDefaultLocator().Append(HdLightTokens->diffuse),
+        HdRetainedTypedSampledDataSource<float>::New(0.f)
+    );
+    editor.Set(
+        HdLightSchema::GetDefaultLocator().Append(HdLightTokens->specular),
+        HdRetainedTypedSampledDataSource<float>::New(0.f)
+    );
+
+    prim.dataSource = editor.Finish();
+}
+
+} // end of anonymous namespace
+
+/// This is a filtering scene index that manages lights primitives
+
+namespace FVP_NS_DEF {
+
+LightsManagementSceneIndex::LightsManagementSceneIndex(const HdSceneIndexBaseRefPtr& inputSceneIndex, const PathInterface& pathInterface, const SdfPath& defaultLightPath) 
+    : ParentClass(inputSceneIndex), 
+    InputSceneIndexUtils(inputSceneIndex)
+    ,_defaultLightPath(defaultLightPath)
+    , _pathInterface(pathInterface)
+{
+}
+
+void LightsManagementSceneIndex::SetLightingMode(LightingMode lightingMode) 
+{ 
+    if (_lightingMode == lightingMode){
+        return;
+    }
+
+    _lightingMode = lightingMode;
+    _DirtyAllLightsPrims();
+}
+
+void LightsManagementSceneIndex::_DirtyAllLightsPrims()
+{
+    HdSceneIndexObserver::DirtiedPrimEntries entries;
+    for (const SdfPath& path : HdSceneIndexPrimView(GetInputSceneIndex())) {
+        auto primType = GetInputSceneIndex()->GetPrim(path).primType;
+        if (_IsALight(primType)) {
+            entries.push_back({ path, HdLightSchema::GetDefaultLocator() });
+        }
+    }
+    _SendPrimsDirtied(entries);
+}
+
+bool LightsManagementSceneIndex::_IsDefaultLight(const SdfPath& primPath)const
+{
+    return primPath == _defaultLightPath;
+}
+
+HdSceneIndexPrim LightsManagementSceneIndex::GetPrim(const SdfPath& primPath) const 
+{
+    auto prim = GetInputSceneIndex()->GetPrim(primPath);
+    auto primType = prim.primType;
+     if (! _IsALight(primType)) {
+         return prim;//return any non light primitive
+     }
+     
+     //This is a light
+     switch (_lightingMode) {
+         case LightingMode::kNoLighting: {
+             _DisableLight(prim);
+             return prim;
+         } break;
+         default:
+         case LightingMode::kSceneLighting: {
+             return prim;
+         } break;
+         case LightingMode::kDefaultLighting: {
+             if (! _IsDefaultLight(primPath)){
+                 _DisableLight(prim);
+             }
+             return prim;//This is also handled in renderOverride.cpp with the _hasDefaultLighting member
+         } break;
+         case LightingMode::kSelectedLightsOnly: {
+             const Ufe::GlobalSelection::Ptr& globalUfeSelection = Ufe::GlobalSelection::get();
+             const Ufe::Selection&            ufeSelection = *globalUfeSelection;
```

Comment:
Minor and optional (depends on personal preference really): since you don't use globalUfeSelection again, may as well just do
```suggestion
              const Ufe::Selection&            ufeSelection = *Ufe::GlobalSelection::get();
```